### PR TITLE
Judge llm analysis implmentation

### PR DESCRIPTION
**Pull request by nemerna**

### CODE REVIEW

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 1]
```diff
@@ -32,14 +56,53 @@ async def judge_llm_analysis(
...
+        if tracker is None:
+            raise ValueError("Tracker must not be None")
+            
         logger.info("Running Judge_LLM_Analysis node - performing LLM analysis")
         logger.info(f"Judge_LLM_Analysis node processing tracker with {len(tracker.issues)} issues")
         
-        # TODO: Implement actual LLM analysis logic here
+        # Initialize LLM service if config is available
+        llm_service = None
+        if tracker.config is not None:
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:69]: What if tracker.config is None? I think it should raise an error or some warning.
WDYT?

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 2]
```diff
@@ -32,14 +56,53 @@ async def judge_llm_analysis(
...
+            try:
+                llm_service = LLMService(tracker.config)
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:71]: We want to use the LLM that is defined in the `config.yml `file, leveraging the AIQ framework, and then pass it to the `IssueAnalysisService` (as you can see in summarize_justifications tool)

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 3]
```diff
@@ -32,14 +56,53 @@ async def judge_llm_analysis(
...
+                logger.error(f"Failed to initialize LLM service: {e}")
+                raise RuntimeError(f"LLM service initialization failed: {e}") from e
+        
+        for issue_id, per_issue in tracker.issues.items():
+            if not isinstance(per_issue, PerIssueData):
+                logger.warning(f"Skipping issue {issue_id}: unexpected data type {type(per_issue)}")
+                continue
+                
+            # Skip if analysis is already final
+            if per_issue.analysis_response and per_issue.analysis_response.is_final != FALSE:
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:82]:
```suggestion
            if per_issue.analysis_response and per_issue.analysis_response.is_final == FinalStatus.TRUE.value:
```

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 4]
```diff
@@ -32,14 +56,53 @@ async def judge_llm_analysis(
...
+                logger.info(f"Skipping issue {issue_id}: analysis already final")
+                continue
+                
+            # Build full analysis context
+            context = _build_analysis_context(per_issue)
+            
+            if llm_service is None:
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:89]: If this skips, won't it skip all issues? Maybe move it after initialization?

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 5]
```diff
@@ -32,14 +56,53 @@ async def judge_llm_analysis(
...
+                logger.warning(f"Skipping analysis for issue {issue_id}: LLM service not available")
+                continue
+                
+            try:
+                # Call primary analysis LLM
+                analysis_response, _ = llm_service.investigate_issue(context, per_issue.issue)
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:95]: This will run the full analysis, including `_recommend` and `summarize_justification`, which are related to other tools. We expect to run only `_analyze_issue_with_retry` in this tool.

**Review by YosiElias — CHANGES_REQUESTED** [review]
Thanks for this PR, @nemerna ! I've left a few comments
pls let me know if they are not clear.

File: src/sast_agent_workflow/tools/judge_llm_analysis.py (hunk 1 above)
Comment (nemerna) [reply src/sast_agent_workflow/tools/judge_llm_analysis.py]: https://github.com/RHEcosystemAppEng/sast-ai-workflow/pull/99/commits/4a2d60a0f43c457945200c3d237cab4c6f0b7143

File: src/sast_agent_workflow/tools/judge_llm_analysis.py (hunk 4 above)
Comment (nemerna) [reply src/sast_agent_workflow/tools/judge_llm_analysis.py]: removed in same commit since we dont rely on the `IssueAnalysisService`
https://github.com/RHEcosystemAppEng/sast-ai-workflow/commit/bc24ef70b38df5f50adb19654bf52a907c9eefb7

File: src/sast_agent_workflow/tools/judge_llm_analysis.py (hunk 5 above)
Comment (nemerna) [reply src/sast_agent_workflow/tools/judge_llm_analysis.py]: same as above resolved as we now rely on `IssueAnalysisService`
https://github.com/RHEcosystemAppEng/sast-ai-workflow/commit/bc24ef70b38df5f50adb19654bf52a907c9eefb7

File: tests/aiq_tests/test_judge_llm_analysis.py [hunk 6]
```diff
@@ -27,47 +27,169 @@ def setUp(self):
...
-           - Non-final vs final issue processing
-           - Analysis response field updates
-           - Edge case tests (empty issues, etc.)
-        """
-        # preparation
-        # TODO: Mock the actual LLM service dependencies when implemented
-        
-        # testing
+    @patch('sast_agent_workflow.tools.judge_llm_analysis.LLMService')
+    async def test_given_sample_tracker_when_judge_llm_analysis_executed_then_processes_non_final_issues(self, mock_llm_service_class):
```
Comment (JudeNiroshan) [inline tests/aiq_tests/test_judge_llm_analysis.py:31]: Please adjust all the test function names according to our convention. Try to make it shorter as well.

File: src/sast_agent_workflow/tools/judge_llm_analysis.py (hunk 2 above)
Comment (YosiElias) [reply src/sast_agent_workflow/tools/judge_llm_analysis.py]: pls pass `llm_name` to `judge_llm_analysis` in `config.yml`

**Review by JudeNiroshan — APPROVED** [review]
LGTM 🚀

File: src/sast_agent_workflow/tools/judge_llm_analysis.py (hunk 2 above)
Comment (nemerna) [reply src/sast_agent_workflow/tools/judge_llm_analysis.py]: 7bcec3bf54c970f9e7ba4c3da194e7e012d92342

File: src/sast_agent_workflow/tools/judge_llm_analysis.py [hunk 7]
```diff
@@ -32,14 +59,64 @@ async def judge_llm_analysis(
...
+                    prompt=prompt_string,
+                    justifications=llm_response.justifications,
+                    short_justifications="",
+                    recommendations=[],
+                    instructions=[],
+                    evaluation=[]
+                )
+                
+                # Update the per-issue analysis response
+                per_issue.analysis_response = analysis_response
```
Comment (YosiElias) [inline src/sast_agent_workflow/tools/judge_llm_analysis.py:112]: Thanks for this change @nemerna !
Can you pls override the relevant fields of the existing per_issue.analysis_response?

I think it's better since it already contains default values for all the fields, and we should keep them until they are populated.

**Review by YosiElias — CHANGES_REQUESTED** [review]
